src/Actions/JoinLines.ts

```diff
@@ -63,6 +63,13 @@ export class ActionJoinLines {
             ranges.forEach(range => {
                 editBuilder.replace(range, ' ');
             });
+
+            // move to the space between the last two joined lines
+            linesToJoin.sort();
+            const lastLine = linesToJoin[linesToJoin.length - 1];
+            const char = activeTextEditor.document.lineAt(lastLine).text.length;
+            const position = new Position(lastLine, char)
```

Comment:
Hmm, I think it it is the correct position ... I've added some passing tests so that you can see.

Reply:
Hmm, seems VSCode is delaying execution of `editBuilder.replace`.
Will take a deeper look into this one.

Reply:
Seems that VSCode is batching all `editBuilder` actions at the end of `activeTextEditor.edit`.

Can we move the replacement part to the end of this function so it's easier to understand?

```ts
// move to the space between the last two joined lines
linesToJoin.sort();
const lastLine = linesToJoin[linesToJoin.length - 1];
const char = activeTextEditor.document.lineAt(lastLine).text.length;
const position = new Position(lastLine, char)
activeTextEditor.selection = new Selection(position, position);

ranges.forEach(range => {
    editBuilder.replace(range, ' ');
});
```

Besides, `sort` is a `O(n*log(n))` operation. We can use a `O(n)` operation to find only the largest line number we want. But this may not be a problem if we are going to deal with multi-cursor.